src/com/puppetlabs/puppetdb/config.clj

```diff
@@ -208,13 +211,20 @@
     (s/validate command-processing-out converted-config)
     (assoc config :command-processing converted-config)))
 
+(defn configure-puppetdb
+  "Validates the [puppetdb] section of the config"
+  [{:keys [puppetdb] :as config :or {puppetdb {}}}]
```

Comment:
I think defaulting to an empty map is pretty good, since it removes the (irrelevant, to us) distinction between an absent config section and an empty section. In any case, this is how it's done elsewhere in stable, and its refactored into `convert-config` in master, with similar defaulting behavior.